virttest/utils_test/__init__.py

```diff
@@ -831,35 +844,47 @@ def install_avocado(installtype="pip"):
         if (session.cmd_status("which avocado") == 0) and not reinstall:
             return True
         if "pip" in installtype:
-            cmd = "python -m pip --version || python -c \"import os; import sys;"
-            cmd += " from six.moves import urllib;" \
-                   "f = urllib.request.urlretrieve(\'https://bootstrap.pypa.io/get-pip.py\')[0]; "
-            cmd += "os.system(\'%s %s\' % (sys.executable, f))\""
-            if session.cmd_status(cmd) > 0:
-                logging.error("pip installation failed")
+            python = find_python(session)
+            if not python:
+                logging.error("Unable to find python.")
                 return False
-            cmd = "pip install avocado-framework"
+
+            cmd = "%s -m pip --version" % python
             if session.cmd_status(cmd) > 0:
-                logging.error("Avocado pip installation failed")
+                logging.debug("pip is not found. Attempt to install it.")
+                cmd = "%s -c \"import os; import sys;" % python
+                cmd += " from six.moves import urllib;" \
+                       "f = urllib.request.urlretrieve(\'https://bootstrap.pypa.io/get-pip.py\')[0]; "
+                cmd += "os.system(\'%s %s\' % (sys.executable, f))\""
+                status, output = session.cmd_status_output(cmd)
+                if status > 0:
+                    logging.error("pip installation failed:\n%s" % output)
+                    return False
+            pip_install_cmd = "%s -m pip -q install" % python
```

Comment:
do you have special reason for quiet (-q)? It was not used before and IMO it's better to have verbose output in case of a failure.

Reply:
No special reason other than not produce a too verbose output. But I understand your point, I am going to change it as well.